### ISSUE TITLE
BigInt: Index out of range: index must not be negative: -1

### ISSUE THREAD

**Issue by modulovalue** [dart-lang/sdk#41819]
I don't know when this happens, and I don't have a repro yet, but this shouldn't throw like that. I'll provide a repro as soon as I have one.
```
(index): Index out of range: index must not be negative: -1 #0      Uint32List.[] (dart:typed_data-patch/typed_data_patch.dart:2493:7)
#1      _BigIntImpl.toDouble (dart:core-patch/bigint_patch.dart:2413:24)
#2      _BigIntImpl./ (dart:core-patch/bigint_patch.dart:1696:43)
#3      Rational.toDouble (package:rational/rational.dart:261:34)
#4      Decimal.toDouble (package:decimal/decimal.dart:170:34)
```

**Comment by modulovalue** [comment]
Repro:
```dart
void main() {
  final a = BigInt.parse("-3696164633313280000000000000");
  final b = BigInt.parse("3481750906500548125063");
  print("${a / b}"); // RangeError (index): Index out of range: index must not be negative: -1
}
```

**Comment by rakudrama** [comment]
Thanks for the report.
The bug exists on all platforms.

```dart
          for (int i = digitIndex; digitIndex >= 0; i--) {
            if (_digits[i] != 0) {
              roundUp();
              break;
            }
          }
```

`digitIndex >= 0` should be `i >= 0`.

**Comment by rakudrama** [comment]
Working on a fix: https://dart-review.googlesource.com/c/sdk/+/147349

**Comment by rakudrama** [comment]
Fixed via 1924d643516d51bdd1736b62919a26ab6b3a41be
